### PR TITLE
better lowhop and fullbright

### PR DESCRIPTION
**Pull request by Turkeii**

### CODE REVIEW

**Comment by Thekingpenguins** [comment]
fuck you turakan lmfao

**Comment by Intoprelised** [comment]
turkey yummy 🤗

**Comment by Turkeii** [comment]
> 
> 
> fuck you turakan lmfao

fan

File: Horion/Module/Modules/Bhop.cpp [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
+	registerEnumSetting("Bhop modes", &Bhopmode, 0);
+	Bhopmode = SettingEnum(this)
+				   .addEntry(EnumEntry("LowhopV2", 0))
+				   .addEntry(EnumEntry("Lowhop", 1))
+				   .addEntry(EnumEntry("Normal", 2));
 	registerFloatSetting("Speed", &this->speed, this->speed, 0.1f, 0.8f);
+	registerFloatSetting("Up", &this->up, this->up, 0.1f, 0.6f);
+	registerFloatSetting("Boost Amount", &this->Boostnum, this->Boostnum, 0.1f, 0.6f);
+	this->registerBoolSetting("Step", &this->step, this->step);
+	this->registerBoolSetting("Boost", &this->Boost, this->Boost);
```
Comment (richardletshacks) [inline Horion/Module/Modules/Bhop.cpp:13]: just check for boost = 0

File: Horion/Module/Modules/Bhop.h [hunk 2]
```diff
@@ -4,13 +4,23 @@
 
 class Bhop : public IModule {
 private:
+	SettingEnum Bhopmode;
```
Comment (richardletshacks) [inline Horion/Module/Modules/Bhop.h:7]: not following naming conventions

File: Horion/Module/Modules/Bhop.h [hunk 3]
```diff
@@ -4,13 +4,23 @@
...
 	float speed = 0.325f;
+	float up = 0.10f;
+	float Boostnum = 0.10f;
```
Comment (richardletshacks) [inline Horion/Module/Modules/Bhop.h:10]: not following naming conventions here either

File: Horion/Module/Modules/Bhop.cpp [hunk 4]
```diff
@@ -1,7 +1,16 @@
 #include "Bhop.h"
 
 Bhop::Bhop() : IModule(0, Category::MOVEMENT, "Hop around like a bunny!") {
+	registerEnumSetting("Bhop modes", &Bhopmode, 0);
+	Bhopmode = SettingEnum(this)
+				   .addEntry(EnumEntry("LowhopV2", 0))
```
Comment (richardletshacks) [inline Horion/Module/Modules/Bhop.cpp:6]: name it something else, no idea what the difference between lowhop and lowhopv2 is

File: Horion/Module/Modules/Bhop.cpp [hunk 5]
```diff
@@ -10,30 +19,92 @@ Bhop::~Bhop() {
...
-
+float getspeed;
+void Bhop::onEnable() {
+	if (Boost)
+	getspeed = speed;
+	if (g_Data.getLocalPlayer() == nullptr)
+		this->setEnabled(false);
+}
 void Bhop::onMove(C_MoveInputHandler* input) {
+	switch (Bhopmode.selected) {
```
Comment (richardletshacks) [inline Horion/Module/Modules/Bhop.cpp:30]: wtf?

File: Horion/Module/Modules/Bhop.cpp [hunk 6]
```diff
@@ -10,30 +19,92 @@ Bhop::~Bhop() {
...
+		lowhopv2 = false;
+		lowhop = false;
+		normal = true;
+		break;
+	}
 	auto player = g_Data.getLocalPlayer();
 	if (player == nullptr) return;
 
-	if (player->isInLava() == 1 || player->isInWater() == 1) 
+	if (player->isInLava() == 1 || player->isInWater() == 1)
```
Comment (richardletshacks) [inline Horion/Module/Modules/Bhop.cpp:50]: == 1 is redundant

File: Horion/Module/Modules/Bhop.cpp [hunk 7]
```diff
@@ -10,30 +19,92 @@ Bhop::~Bhop() {
...
 
 	vec2_t moveVec2d = {input->forwardMovement, -input->sideMovement};
 	bool pressed = moveVec2d.magnitude() > 0.01f;
-
-	if (player->onGround && pressed)
+	bool dohop = false;
+	if (pressed && player->onGround) {
+		dohop = true;
+	} else
+		dohop = false;
```
Comment (richardletshacks) [inline Horion/Module/Modules/Bhop.cpp:62]: just do ``dohop = pressed && player->onGround``

**Review by richardletshacks — CHANGES_REQUESTED** [review]
the speed thing is just so messed up I don't wanna continue the review anymore, I suggest making seperate pull requests and completely rewriting the speed

**Comment by richardletshacks** [comment]
closed for now, feel free to make a new pull request when the requested changes have been made

**Pull request closed without merging**